### PR TITLE
Added Middle Of LinkedList in Java

### PR DESCRIPTION
**Pull request by Priya730**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added the code to find the Middle Element of a LinkedList in JAVA. 

### Checklist

- [ ] I've read the contribution guidelines.
- [ ] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes issue #2337

### CODE REVIEW

**Comment by Priya730** [comment]
@HarshCasper It would be very helpful if you could provide feedback and review the PR. Thanks!

File: Java/ds/MiddleOfLinkedList.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        MiddleOfLinkedList list = new MiddleOfLinkedList();
+        
+        // Creating a linked list
+        Node head = new Node(1);
+        list.addToTheLast(head);
+        list.addToTheLast(new Node(2));
+        list.addToTheLast(new Node(3));
+        list.addToTheLast(new Node(4));
+        list.addToTheLast(new Node(5));
+
```
Comment (HarshCasper) [inline Java/ds/MiddleOfLinkedList.java]: Make the User Input dynamic

File: Java/ds/MiddleOfLinkedList.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
+package linkedList;
+//Java program to find middle of LinkedList
+
+/*Possible Approaches: 
+					   i) Traverse List: Traverse list to find it's length, then scan list to locate 							n/2 node from the beginning.
+					   ii) Two - Pointer Approach: It is the efficient approach (only one scan). 
+					   		Use 2 pointers. Move one at twice the speed of second. 
+					   		When first pointer reaches the end of list, second pointer will point the 							middle node of the LinkedList.
+Using the 2 pointer approach to find the middle element with Time Complexity: O(n) Space Complexity: O(1)					   		
+**/
```
Comment (HarshCasper) [inline Java/ds/MiddleOfLinkedList.java]: Left-Align the Content

File: Java/ds/MiddleOfLinkedList.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
+package linkedList;
```
Comment (HarshCasper) [inline Java/ds/MiddleOfLinkedList.java]: Remove the package

File: Java/ds/MiddleOfLinkedList.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        
+        System.out.println("Middle node is: "+ middle.data); //Prints middle element
+ 
+    }
+}
+
+/*OUTPUT
+  1 2 3 4 5 
+  Middle node is: 3
+ */
```
Comment (HarshCasper) [inline Java/ds/MiddleOfLinkedList.java:92]: Add new line at the EOF

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Java/ds/MiddleOfLinkedList.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
+//Java program to find middle of LinkedList
+
+/*Possible Approaches: 
+i) Traverse List: Traverse list to find it's length, then scan list to locate n/2 node from the beginning.
+ii) Two - Pointer Approach: It is the efficient approach (only one scan). Use 2 pointers. Move one at twice the speed of second. 
+When first pointer reaches the end of list, second pointer will point the middle node of the LinkedList.
+
+Using the 2 pointer approach to find the middle element with Time Complexity: O(n) Space Complexity: O(1)					   		
+**/
```
Comment (ankitaggarwal23) [inline Java/ds/MiddleOfLinkedList.java]: Remove this line as you already mentioned at the EOF of the code and Extra one *

File: Java/ds/MiddleOfLinkedList.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+public class MiddleOfLinkedList {
```
Comment (ankitaggarwal23) [inline Java/ds/MiddleOfLinkedList.java:9]: Missing java header file

File: Java/ds/MiddleOfLinkedList.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+	Node head;
+	 
+    static class Node {
+    	int data;
+        Node next;
+     
+        // constructor
```
Comment (ankitaggarwal23) [inline Java/ds/MiddleOfLinkedList.java]: Remove redundant comments

File: Java/ds/MiddleOfLinkedList.java [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+    			i=1;
+    		}
+
+    		else if(i == 1){
+    			pointer1 = pointer1.next;  //Increment both the pointers
+    			pointer2 = pointer2.next;
+    			i=0;
+    		}
+    	}	
+
```
Comment (ankitaggarwal23) [inline Java/ds/MiddleOfLinkedList.java]: Remove all redundant comments and align your code properly

File: Java/ds/MiddleOfLinkedList.java [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+    		else if(i == 1){
+    			pointer1 = pointer1.next;  //Increment both the pointers
+    			pointer2 = pointer2.next;
+    			i=0;
+    		}
+    	}	
+    	
+    	return pointer2;  //Return pointer2 which now points to the middle node
+ 
+    }
```
Comment (ankitaggarwal23) [inline Java/ds/MiddleOfLinkedList.java]: Remove redundant comments and extra line

File: Java/ds/MiddleOfLinkedList.java [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+    	
+    //Main Method
+    public static void main(String[] args) {
+        MiddleOfLinkedList list = new MiddleOfLinkedList();
+        
+        // Creating a linked list
```
Comment (ankitaggarwal23) [inline Java/ds/MiddleOfLinkedList.java]: Remove redundant comments

File: Java/ds/MiddleOfLinkedList.java [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+        Scanner sc = new Scanner(System.in);
+        
+        //size of linked list
```
Comment (ankitaggarwal23) [inline Java/ds/MiddleOfLinkedList.java]: Remove redundant comments

File: Java/ds/MiddleOfLinkedList.java [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+        System.out.println("Enter the size of Linked List:");
+        int n = sc.nextInt();
+        System.out.println("Enter the data of Linked List:");
+        //Input for the Linked list
```
Comment (ankitaggarwal23) [inline Java/ds/MiddleOfLinkedList.java]: Remove redundant comments

File: Java/ds/MiddleOfLinkedList.java [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+        int num = sc.nextInt();
+        Node head = new Node(num);
+        list.addToTheLast(head);
+        Node tail = head;
+        //Linked List created
```
Comment (ankitaggarwal23) [inline Java/ds/MiddleOfLinkedList.java]: Remove redundant comments

File: Java/ds/MiddleOfLinkedList.java [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+        for(int i=0; i<n-1; i++)
+        {
+            num = sc.nextInt();
+            tail.next = new Node(num);
+            tail = tail.next;
+        }
+ 
+        list.printList();  //prints list
```
Comment (ankitaggarwal23) [inline Java/ds/MiddleOfLinkedList.java]: Remove redundant comments

File: Java/ds/MiddleOfLinkedList.java [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+        
+        // finding the middle element of the LinkedList
+        Node middle = list.findMiddleNode(head);
+        
+        System.out.println("Middle node is: "+ middle.data); //Prints middle element
+ 
+    }
```
Comment (ankitaggarwal23) [inline Java/ds/MiddleOfLinkedList.java]: Remove redundant comments and extra line

File: Java/ds/MiddleOfLinkedList.java [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+
+/*OUTPUT
+  Enter the size of Linked List:
+  5
+  Enter the data of Linked List:
+  1 2 3 4 5
+  1 2 3 4 5 
+  Middle node is: 3
+  
+  Time Complexity: O(n) Space Complexity: O(1)
```
Comment (ankitaggarwal23) [inline Java/ds/MiddleOfLinkedList.java]: Shift Space Complexity to new line

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**